### DISCUSSION TITLE
Lesson 12. Quick question about on Handler mintDsc

### DISCUSSION THREAD

**Discussion by DownShiftDom** [Cyfrin/foundry-full-course-cu#516] [Q&A] [answered]
```solidity
function mintDsc(uint256 amount) public {
        (uint256 totalDscMinted, uint256 collateralValueInusd) = dsce.getAccountInformation(msg.sender);

        uint256 maxDscToMint = (collateralValueInusd / 2) - totalDscMinted;
        if (maxDscToMint < 0) {
            return;
        }
        amount = bound(amount, 0, uint256(maxDscToMint));
        if (amount == 0) {
            return;
        }

        vm.startPrank(msg.sender);
        dsce.mintDsc(amount);
        vm.stopPrank();
    }

```

Can someone explain why in the following code 
```solidity 
uint256 maxDscToMint = (collateralValueInusd / 2) - totalDscMinted;
``` 
we divide collateralValueInUsd by 2? Sorry if this is a silly question. Its currently 1am for me and my brain is malfunctioning lol

**Comment by 0xusmanf** [comment, marked as answer]
Suppose you deposited 100 USD worth of collateral which means you can only mint DSC half of the value of your collateral because protocol is 200% collateralized. So to determine the amount of DSC you can mint you divide it by 2 -> `(collateralValueInusd / 2)` and then you minus the amount of DSC you already minted -> `(collateralValueInusd / 2) - totalDscMinted`.

**Reply by DownShiftDom** [reply]
Thank you for clearing that up! forgot about the liquidation threshold! oops.
